File: src/VBNetFx/FacessoData/DataClasses/ParameterClasses/ShiftParameters.vb
''' <summary>
''' Hält die Schichteinstellungen (Schichten, Alternierende Schritte, etc.) für Auswertungen
''' </summary>
''' <remarks></remarks>
<Serializable()> _
Public Class ShiftParameters

    Private myConsiderShift1 As Boolean
    Private myConsiderShift2 As Boolean
    Private myConsiderShift3 As Boolean
    Private myConsiderShift4 As Boolean
    Private myAlternateShifts As Boolean
    Private myDaysAfterToAlternate As Integer
    Private myAlternatingFirstShift As Integer
    Private myAlternatingSecondShift As Integer

    Sub New()
        MyBase.New()

        'Kritische Default-Werte füllen (können bei der UI-Befüllung in die Hose gehen)
        myAlternateShifts = False
        myAlternatingFirstShift = 1
        myAlternatingSecondShift = 2
        myDaysAfterToAlternate = 7
    End Sub

    Sub New(ByVal CShift1 As Boolean, ByVal CShift2 As Boolean, ByVal CShift3 As Boolean, ByVal CShift4 As Boolean, _
            ByVal AlternateShifts As Boolean, ByVal DaysAfterToAlternate As Integer, _
            ByVal AlternatingFirstShift As Integer, ByVal AlternatingSecondShift As Integer)
        myConsiderShift1 = CShift1
        myConsiderShift2 = CShift2
        myConsiderShift3 = CShift3
        myConsiderShift4 = CShift4
        myAlternateShifts = AlternateShifts
        myDaysAfterToAlternate = DaysAfterToAlternate
        myAlternatingFirstShift = AlternatingFirstShift
        myAlternatingSecondShift = AlternatingSecondShift
    End Sub

    Public Property ConsiderShift1() As Boolean
        Get
            Return myConsiderShift1
        End Get
        Set(ByVal value As Boolean)
            myConsiderShift1 = value
        End Set
    End Property

    Public Property ConsiderShift2() As Boolean
        Get
            Return myConsiderShift2
        End Get
        Set(ByVal value As Boolean)
            myConsiderShift2 = value
        End Set
    End Property

    Public Property ConsiderShift3() As Boolean
        Get
            Return myConsiderShift3
        End Get
        Set(ByVal value As Boolean)
            myConsiderShift3 = value
        End Set
    End Property

    Public Property ConsiderShift4() As Boolean
        Get
            Return myConsiderShift4
        End Get
        Set(ByVal value As Boolean)
            myConsiderShift4 = value
        End Set
    End Property

    Public Property AlternateShifts() As Boolean
        Get
            Return myAlternateShifts
        End Get
        Set(ByVal value As Boolean)
            myAlternateShifts = value
        End Set
    End Property

    Public Property DaysAfterToAlternate() As Integer
        Get
            Return myDaysAfterToAlternate
        End Get
        Set(ByVal value As Integer)
            myDaysAfterToAlternate = value
        End Set
    End Property

    Public Property AlternatingFirstShift() As Integer
        Get
            Return myAlternatingFirstShift
        End Get
        Set(ByVal value As Integer)
            myAlternatingFirstShift = value
        End Set
    End Property

    Public Property AlternatingSecondShift() As Integer
        Get
            Return myAlternatingSecondShift
        End Get
        Set(ByVal value As Integer)
            myAlternatingSecondShift = value
        End Set
    End Property

    Public Overrides Function ToString() As String

        If AlternateShifts Then
            Return "Wechselschicht zwischen Schicht " & AlternatingFirstShift & " und Schicht " & AlternatingSecondShift & " alle " & DaysAfterToAlternate & " Tage."
        End If

        Dim locShiftCol As New ArrayList
        If ConsiderShift1 Then
            locShiftCol.Add(1)
        End If
        If ConsiderShift2 Then
            locShiftCol.Add(2)
        End If
        If ConsiderShift3 Then
            locShiftCol.Add(3)
        End If

        Dim locRetString As String = "Schicht "
        For c As Integer = 0 To locShiftCol.Count - 1
            locRetString &= locShiftCol(c).ToString
            If c < (locShiftCol.Count - 2) Then
                locRetString &= ", "
            End If
            If c = (locShiftCol.Count - 2) And locShiftCol.Count > 1 Then
                locRetString &= " und "
            End If
        Next
        Return locRetString
    End Function
End Class

File: src/MixNetFx/FacessoData/DataClasses/ParameterClasses/ShiftParameters.cs
using System;
using System.Collections;

namespace Facesso.Data
{
    /// <summary>
    /// Hält die Schichteinstellungen (Schichten, Alternierende Schritte, etc.) für Auswertungen
    /// </summary>
    [Serializable]
    public class ShiftParameters
    {
        private bool myConsiderShift1;
        private bool myConsiderShift2;
        private bool myConsiderShift3;
        private bool myConsiderShift4;
        private bool myAlternateShifts;
        private int myDaysAfterToAlternate;
        private int myAlternatingFirstShift;
        private int myAlternatingSecondShift;

        public ShiftParameters()
        {
            myAlternateShifts = false;
            myAlternatingFirstShift = 1;
            myAlternatingSecondShift = 2;
            myDaysAfterToAlternate = 7;
        }

        public ShiftParameters(bool cShift1, bool cShift2, bool cShift3, bool cShift4,
            bool alternateShifts, int daysAfterToAlternate,
            int alternatingFirstShift, int alternatingSecondShift)
        {
            myConsiderShift1 = cShift1;
            myConsiderShift2 = cShift2;
            myConsiderShift3 = cShift3;
            myConsiderShift4 = cShift4;
            myAlternateShifts = alternateShifts;
            myDaysAfterToAlternate = daysAfterToAlternate;
            myAlternatingFirstShift = alternatingFirstShift;
            myAlternatingSecondShift = alternatingSecondShift;
        }

        public bool ConsiderShift1
        {
            get { return myConsiderShift1; }
            set { myConsiderShift1 = value; }
        }

        public bool ConsiderShift2
        {
            get { return myConsiderShift2; }
            set { myConsiderShift2 = value; }
        }

        public bool ConsiderShift3
        {
            get { return myConsiderShift3; }
            set { myConsiderShift3 = value; }
        }

        public bool ConsiderShift4
        {
            get { return myConsiderShift4; }
            set { myConsiderShift4 = value; }
        }

        public bool AlternateShifts
        {
            get { return myAlternateShifts; }
            set { myAlternateShifts = value; }
        }

        public int DaysAfterToAlternate
        {
            get { return myDaysAfterToAlternate; }
            set { myDaysAfterToAlternate = value; }
        }

        public int AlternatingFirstShift
        {
            get { return myAlternatingFirstShift; }
            set { myAlternatingFirstShift = value; }
        }

        public int AlternatingSecondShift
        {
            get { return myAlternatingSecondShift; }
            set { myAlternatingSecondShift = value; }
        }

        public override string ToString()
        {
            if (AlternateShifts)
                return "Wechselschicht zwischen Schicht " + AlternatingFirstShift + " und Schicht " + AlternatingSecondShift + " alle " + DaysAfterToAlternate + " Tage.";

            var locShiftCol = new ArrayList();
            if (ConsiderShift1) locShiftCol.Add(1);
            if (ConsiderShift2) locShiftCol.Add(2);
            if (ConsiderShift3) locShiftCol.Add(3);

            string locRetString = "Schicht ";
            for (int c = 0; c < locShiftCol.Count; c++)
            {
                locRetString += locShiftCol[c].ToString();
                if (c < (locShiftCol.Count - 2))
                    locRetString += ", ";
                if (c == (locShiftCol.Count - 2) && locShiftCol.Count > 1)
                    locRetString += " und ";
            }
            return locRetString;
        }
    }
}
